src/domain/account/getNfts.js

```diff
@@ -3,11 +3,22 @@ const NFT = require('../../infra/nft');
 
 const nftInstance = new NFT();
 
-async function getNfts({ address, search, chain }) {
-  const nfts = await nftInstance.getOwnedNFTs({
-    address,
-    chain: chain || config.CHAIN,
-  });
+async function getNfts({ address, search, chain, tokenAddress, tokenId }) {
+  let nfts = [];
+
+  if (tokenAddress && tokenId) {
+    nfts = await nftInstance.getNftByMetadata({
+      chain: chain,
```

Comment:
Instead you should get Owned NFTs and filter them by metadata

Reply:
we can't do that, because that'll be lot of traversal. For instance, our test user who raise the issue had 500+ nfts.